services/src/api/handlers/machine_learning.rs

```diff
@@ -60,12 +66,27 @@ pub(crate) async fn add_ml_model<C: ApplicationContext>(
     app_ctx: web::Data<C>,
     model: web::Json<MlModel>,
 ) -> Result<web::Json<MlModelNameResponse>, MachineLearningError> {
+    let session_context = app_ctx.session_context(session);
+    let exe_context = session_context
+        .execution_context()
+        .expect("Execution Context must exist");
+
     let model = model.into_inner();
-    let id_and_name = app_ctx
-        .session_context(session)
-        .db()
-        .add_model(model)
-        .await?;
+
+    // This call also checks that the file is available!
+    let ml_model_metadata = model.metadata_for_operator()?;
+    // Check that the in/out shapes are ok
+    check_model_shape(
+        &ml_model_metadata,
+        exe_context.tiling_specification().tile_size_in_pixels,
+    )?;
+    // initialize model
+    // TODO: re-use session accross queries?
```

Comment:
there is no query happening? coment copy pasted?

Reply:
[e936c08](https://github.com/geo-engine/geoengine/pull/1000/commits/e936c087dee32a3a244f7a2dc6972fda4787f8d6)